cryptocurrency-realtime-trend/src/main/java/JetCoinTrend.java

```diff
@@ -0,0 +1,95 @@
+        Vertex relevance = dag.newVertex("relevance", of(RelevanceProcessor::new));
+        Vertex sentiment = dag.newVertex("sentiment", of(SentimentProcessor::new));
+
+        WindowDefinition slidingWindowDef1Min = WindowDefinition.slidingWindowDef(10000, 5000);
+        WindowDefinition slidingWindowDef5Min = WindowDefinition.slidingWindowDef(300000, 60000);
+        WindowDefinition slidingWindowDef15Min = WindowDefinition.slidingWindowDef(900000, 300000);
+
+        WatermarkGenerationParams<TimestampedEntry<String, Double>> params = WatermarkGenerationParams.wmGenParams((DistributedToLongFunction<TimestampedEntry<String, Double>>) TimestampedEntry::getTimestamp, withFixedLag(5000), emitByFrame(slidingWindowDef1Min), 60000);
+        DistributedSupplier<Processor> insertWMP = insertWatermarksP(params);
+
+        Vertex insertWm = dag.newVertex("insertWm", insertWMP).localParallelism(1);
+
+        Vertex aggregateTrend1Min = dag.newVertex("aggregateTrend1Min", aggregateToSlidingWindowP(TimestampedEntry::getKey,
+                TimestampedEntry::getTimestamp,
+                TimestampKind.EVENT,
+                slidingWindowDef1Min, averagingDouble((DistributedToDoubleFunction<TimestampedEntry<String, Double>>) TimestampedEntry::getValue)
+        ));
+
+        Vertex aggregateTrend5Min = dag.newVertex("aggregateTrend5Min", aggregateToSlidingWindowP(TimestampedEntry::getKey,
+                TimestampedEntry::getTimestamp,
+                TimestampKind.EVENT,
+                slidingWindowDef5Min, averagingDouble(
+                        (DistributedToDoubleFunction<TimestampedEntry<String, Double>>) TimestampedEntry::getValue)
+        ));
+
+        Vertex aggregateTrend15Min = dag.newVertex("aggregateTrend15Min", aggregateToSlidingWindowP(TimestampedEntry::getKey,
+                TimestampedEntry::getTimestamp,
+                TimestampKind.EVENT,
+                slidingWindowDef15Min, averagingDouble(
+                        (DistributedToDoubleFunction<TimestampedEntry<String, Double>>) TimestampedEntry::getValue)
+        ));
+
+
+        Vertex sink = dag.newVertex("sink", DiagnosticProcessors.writeLoggerP());
+
+        dag.edge(Edge.between(twitterSource, insertWm));
+//        dag.edge(Edge.between(redditSource, insertWm));
+        dag.edge(Edge.between(insertWm, relevance));
+        dag.edge(Edge.between(relevance, sentiment));
+        dag.edge(Edge.from(sentiment, 0).to(aggregateTrend1Min).partitioned(
+                (DistributedFunction<TimestampedEntry<String, Double>, Object>) TimestampedEntry::getKey));
+        dag.edge(Edge.from(sentiment, 1).to(aggregateTrend5Min).partitioned(
+                (DistributedFunction<TimestampedEntry<String, Double>, Object>) TimestampedEntry::getKey));
+        dag.edge(Edge.from(sentiment, 2).to(aggregateTrend15Min).partitioned(
+                (DistributedFunction<TimestampedEntry<String, Double>, Object>) TimestampedEntry::getKey));
```

Comment:
since it's calculating 3 windows and they're multiple of each other, it would be much more efficient to do this as cascading windows.

for example, you use the output of the 5sec window to calculate 10sec, 15sec and so on. Rather than have an edge from the processor to each sliding window, you would instead feed the next sliding window from the previous one.